source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRUILabel.mm

```diff
@@ -57,4 +57,21 @@ - (void)layoutSubviews
     _area = area;
 }
 
+- (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
+{
+    CGPoint location = point;
+    location.x -= self.textContainerInset.left;
+    location.y -= self.textContainerInset.top;
+    CGFloat fraction = 0.0f;
+    NSUInteger characterIndex = [self.layoutManager characterIndexForPoint:location
+                                                           inTextContainer:self.textContainer
+                                  fractionOfDistanceBetweenInsertionPoints:&fraction];
+    if (!(fraction == 0.0 || fraction == 1.0) && characterIndex < self.textStorage.length) {
+        if ([self.textStorage attribute:NSLinkAttributeName atIndex:characterIndex effectiveRange:NULL]) {
+            return self;
+        }
+    }
```

Comment:
I'm sorry, I don't understand the intention of this change. We're ignoring UI interactions unless they land on a character that's part of a hyperlink?

But only if they land inside the character, and not on the left or right edge of it?

Reply:
that's correct. this method is part of UIView API, and it's used for returning the farthest UIView that should receive the given UIEvent in UIView tree.
This UIView is subclass of UITextView, and only UIEvent it should handle is touch event on hyperlink. This change is needed because our container classes should receive AdaptiveCards events, and the textview should handle touch event  on hyperlink. 

---
In reply to: [226504988](https://github.com/Microsoft/AdaptiveCards/pull/2106#discussion_r226504988) [](ancestors = 226504988)

Reply:
i'm not exactly sure what you meant by "dropping inputs that happen to land exactly between two characters". 

characterIndexForPoint method returns nearest character index in the text storage, so the returned index will be always valid character index.

the fraction tell you how far the touch is from the insertion point. if it's zero or 1, it's outside of characters in line fragment. linefragment is filled with characters, but linefragment will occupy one line of textview, and its horizontal position is determined by LayoutManager. The fraction will have value between 0 and 1.0 when touch lands in the frame of the line fragement. 

---
In reply to: [226796362](https://github.com/Microsoft/AdaptiveCards/pull/2106#discussion_r226796362) [](ancestors = 226796362,226741749,226504988)